packages/dd-trace/src/datastreams/pathway.js

```diff
@@ -0,0 +1,44 @@
+// encoding used here is FNV1a
+// other languages use FNV1
+// this inconsistency is ok because hashes do not need to be consistent across services
+const fnv = require('fnv-plus')
+const { encodeVarint, decodeVarint } = require('./encoding')
+
+function fnvHash (checkpointString) {
+  const hash = fnv.hash(checkpointString, 64)
+  return Buffer.from(hash.hex(), 'hex')
+}
+
+function computeHash (service, env, edgeTags, parentHash) {
+  const currentHash = fnvHash(`${service}${env}` + edgeTags.join(''))
+  const buf = Buffer.concat([ currentHash, parentHash ], 16)
+  return fnvHash(buf.toString())
+}
+
+function encodePathwayContext (dataStreamsContext) {
+  return Buffer.concat([ dataStreamsContext.hash, Buffer.from(encodeVarint(Math.round(dataStreamsContext.pathwayStartNs / 1e6))), Buffer.from(encodeVarint(Math.round(dataStreamsContext.edgeStartNs / 1e6))) ], 20)
```

Comment:
This would be more readable if split in multiple lines.

Reply:
```js
  return Buffer.concat([
    dataStreamsContext.hash,
    Buffer.from(encodeVarint(Math.round(dataStreamsContext.pathwayStartNs / 1e6))),
    Buffer.from(encodeVarint(Math.round(dataStreamsContext.edgeStartNs / 1e6)))
  ], 20)
```